shared/constants/platform.desktop.js

```diff
@@ -56,10 +56,19 @@ function win32SocketDialPath (): string {
 }
 
 function linuxSocketDialPath (): string {
-  if (runMode === 'prod') {
-    return path.join(`${getenv('XDG_RUNTIME_DIR', '')}/keybase/`, socketName)
+  // If XDG_RUNTIME_DIR is defined use that, else use $HOME/.config.
+  const homeDir = getenv('HOME', '')
+  const homeCacheDir = path.join(homeDir, '.config')
+  const runtimeDir = getenv('XDG_RUNTIME_DIR', '')
+
+  const cacheDir = runtimeDir || homeCacheDir
+  const suffix = runMode === 'prod' ? '' : `.${runMode}`
+
+  if (!runtimeDir && !homeDir) {
+    console.warn("You don't have $HOME or $XDG_RUNTIME_DIR defined, so we can't find the Keybase service path.")
```

Comment:
We're still proceeding with the empty string in this case, which means the global `socketPath` is going to wind up assuming the current directory is the home. That's probably true in most cases for desktop autostarts, but it's frequently not true if I run `run_keybase` in some random terminal, and it's going to cause confusing bugs in that case. Do we have any path right now for reporting fatal errors that happen while the app is starting? This should take that path if so, I think. If not -- meh, whatever, probably no one is actually missing the `HOME` var :p